pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -70,6 +71,19 @@ func getResourceLabels(app *v1beta2.SparkApplication) map[string]string {
 	return labels
 }
 
+func getResourceAnnotations(app *v1beta2.SparkApplication) map[string]string {
+	annotations := map[string]string{}
+
+	if app.Spec.SparkConf["spark.ui.ingress.class"] == "nginx" {
```

Comment:
Instead of defining these Spark config options, how about the following: make a struct type field that encapsulates all UI related config options, e.g.,:

```
type UIConfig struct {
  IngressClass *string `json:"ingressClass,omitempty"`
  EnableSSL bool `json:"enableSSL,omitempty"`
  ForceSSL bool `json:"forceSSL,omitempty"`
}
```

And then add an optional field of type `UIConfig` in `SparkApplicationSpec`.

Reply:
Hello. Thanks for your answer.
Based on your advice, i made this new version, declaring a UIConfig struct.
All my tests are working fine, with or without the use of this struct, and whatever the values of its parameters are. But i only handle the nginx ingress Class because its the most used and the only one i use myself. For those who need to adapt for another class, they will only have to append the sparkapplication.getResourceAnnotations function.